flink-test-utils-parent/flink-test-utils-junit/src/main/java/org/apache/flink/testutils/logging/LoggerAuditingExtension.java

```diff
@@ -81,7 +81,9 @@ public void beforeEach(ExtensionContext context) throws Exception {
                 new AbstractAppender("test-appender", null, null, false, Property.EMPTY_ARRAY) {
                     @Override
                     public void append(LogEvent event) {
-                        loggingEvents.add(event.toImmutable());
+                        if (event.toImmutable() != null) {
+                            loggingEvents.add(event.toImmutable());
+                        }
```

Comment:
It's not clear to me why this change is necessary. The `LogEvent` interface doesn't indicate that `null` is a valid return value. :thinking: You might want to add a comment here if we really need to check for `null` values here.

Reply:
I've seen NPE logged, but don't see it anymore - will remove the check.